packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -48,18 +48,124 @@ export class ArtifactCreator {
         renderer => ({...renderer, file: `${renderer.name}-golden.png`}));
   }
 
-  async captureAndAnalyzeScreenshots(
-      scenarioWhitelist: Set<string>|null = null) {
+  async compareRenderers(scenario: ScenarioConfig) {
+    const analysisResults: AnalysisResults = [];
+    const {rootDirectory, outputDirectory, goldens} = this;
+    const {name: scenarioName, exclude, dimensions} = scenario;
+
+    console.log(
+        `Start to compare model-viewer's golden with other renderers' goldens:`);
+
+    const modelViewerIndex = 0;
+    const modelViewerGoldenPath = join(
+        rootDirectory, 'goldens', scenarioName, goldens[modelViewerIndex].file);
+    let modelViewerGolden;
+    try {
+      modelViewerGolden = await fs.readFile(modelViewerGoldenPath);
+    } catch (error) {
+      throw new Error(`❌ Failed to read model-viewer's ${
+          scenarioName} golden! Error message: ${error.message}`);
+    }
+
+    await fs.writeFile(
+        join(outputDirectory, scenarioName, goldens[modelViewerIndex].file),
+        modelViewerGolden);
+
+    const modelViewerGoldenImage = pngjs.PNG.sync.read(modelViewerGolden).data;
+
+    for (let golden of goldens) {
```

Comment:
I think you can do `const golden of goldens` if you're not actually modifying the elements.